File: src/main/java/com/petcare/product/entity/ProductOrder.java
package com.petcare.product.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.petcare.common.entity.BaseEntity;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * Entity for table `product_order`.
 */
@Getter
@Setter
@NoArgsConstructor
@TableName("product_order")
public class ProductOrder extends BaseEntity {

    @TableField("order_no")
    private String orderNo;

    @TableField("user_id")
    private Long userId;

    @TableField("store_id")
    private Long storeId;

    @TableField("total_amount")
    private BigDecimal totalAmount;

    @TableField("payment_method")
    private String paymentMethod;

    @TableField("payment_status")
    private String paymentStatus;

    @TableField("pickup_status")
    private String pickupStatus;

    @TableField("status")
    private String status;

    @TableField("contact_name")
    private String contactName;

    @TableField("contact_phone")
    private String contactPhone;

    @TableField("remark")
    private String remark;

    @TableField("merchant_remark")
    private String merchantRemark;

    @TableField("confirm_time")
    private LocalDateTime confirmTime;

    @TableField("complete_time")
    private LocalDateTime completeTime;

    @TableField("cancel_time")
    private LocalDateTime cancelTime;
}

File: frontend/admin-web/src/api/product-order.ts
import request from '../utils/request'
import type { PageResponse, PageParams } from '../types/api'

// ─── Types ───

export interface ProductOrder {
  id: number
  orderNo: string
  totalAmount: number
  paymentMethod: string
  paymentStatus: string
  pickupStatus: string
  status: string
  contactName: string
  contactPhone: string
  remark: string | null
  createTime: string
  confirmTime: string | null
  completeTime: string | null
  cancelTime: string | null
}

export interface ProductOrderItem {
  id: number
  productId: number
  productName: string
  productCoverUrl: string
  price: number
  quantity: number
  totalAmount: number
}

export interface ProductOrderDetail extends ProductOrder {
  userId: number
  storeId: number
  merchantRemark: string | null
  items: ProductOrderItem[]
}

export interface ProductOrderQueryParams extends PageParams {
  status?: string
}

export interface AdminOrderActionParams {
  reason?: string
}

// ─── API Functions ───

export const getProductOrderList = (params: ProductOrderQueryParams) => {
  return request.get<PageResponse<ProductOrder>>('/v1/admin/product-orders', { params })
}

export const getProductOrderDetail = (id: number) => {
  return request.get<ProductOrderDetail>(`/v1/admin/product-orders/${id}`)
}

export const confirmProductOrder = (id: number) => {
  return request.post<ProductOrder>(`/v1/admin/product-orders/${id}/confirm`)
}

export const readyProductOrder = (id: number) => {
  return request.post<ProductOrder>(`/v1/admin/product-orders/${id}/ready`)
}

export const confirmPaymentOrder = (id: number) => {
  return request.post<ProductOrder>(`/v1/admin/product-orders/${id}/confirm-payment`)
}

export const completeProductOrder = (id: number) => {
  return request.post<ProductOrder>(`/v1/admin/product-orders/${id}/complete`)
}

export const cancelProductOrder = (id: number, data?: AdminOrderActionParams) => {
  return request.post<ProductOrder>(`/v1/admin/product-orders/${id}/cancel`, data)
}

export function outOfStockProductOrder(id: number, data?: AdminOrderActionParams) {
  return request.post<ProductOrder>(`/v1/admin/product-orders/${id}/out-of-stock`, data)
}
